langchain4j-opensearch/src/test/java/dev/langchain4j/store/embedding/opensearch/OpenSearchEmbeddingStoreAWSTest.java

```diff
@@ -1,38 +1,64 @@
 package dev.langchain4j.store.embedding.opensearch;
 
+import com.github.dockerjava.api.command.InspectContainerResponse;
+import com.jayway.jsonpath.JsonPath;
 import dev.langchain4j.data.segment.TextSegment;
 import dev.langchain4j.model.embedding.AllMiniLmL6V2QuantizedEmbeddingModel;
 import dev.langchain4j.model.embedding.EmbeddingModel;
 import dev.langchain4j.store.embedding.EmbeddingStore;
 import dev.langchain4j.store.embedding.EmbeddingStoreIT;
 import lombok.SneakyThrows;
-import org.junit.jupiter.api.Disabled;
+import net.minidev.json.JSONArray;
+import org.junit.jupiter.api.BeforeAll;
 import org.opensearch.client.transport.aws.AwsSdk2TransportOptions;
-import software.amazon.awssdk.auth.credentials.ProfileCredentialsProvider;
+import org.testcontainers.containers.Container.ExecResult;
+import org.testcontainers.images.builder.Transferable;
+import org.testcontainers.junit.jupiter.Container;
+import org.testcontainers.junit.jupiter.Testcontainers;
+import org.testcontainers.utility.DockerImageName;
+import software.amazon.awssdk.auth.credentials.AwsBasicCredentials;
+import software.amazon.awssdk.auth.credentials.StaticCredentialsProvider;
+
+import java.io.IOException;
+import java.time.Duration;
 
 import static dev.langchain4j.internal.Utils.randomUUID;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.awaitility.Awaitility.await;
 
-@Disabled("Needs OpenSearch running with AWS")
+@Testcontainers
 public class OpenSearchEmbeddingStoreAWSTest extends EmbeddingStoreIT {
 
-    /**
-     * To run the tests locally, you have to provide an Amazon OpenSearch domain. The code uses
-     * the credentials stored locally in your machine. For more information about how to configure
-     * your credentials locally, see https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-files.html.
-     */
+    @Container
+    private static final LocalStackContainer localstack = new LocalStackContainer(DockerImageName.parse("localstack/localstack:latest"));
```

Comment:
using latest due to a recent fix that it is only available in latest. Later should use a fixed version